tests/conftest.py

```diff
@@ -560,23 +342,14 @@ def assert_side_effects_invoked(side_effects_contract, side_effects_trigger, n=1
     return assert_side_effects_invoked
 
 
-@pytest.fixture
-def get_logs(w3):
-    def get_logs(tx_hash, c, event_name):
-        tx_receipt = w3.eth.get_transaction_receipt(tx_hash)
-        return c._classic_contract.events[event_name]().process_receipt(tx_receipt)
-
-    return get_logs
-
-
 @pytest.fixture(scope="module")
-def tx_failed(tester):
+def tx_failed(env):
+    # todo: use custom exception, we don't use eth_tester anymore
     @contextmanager
     def fn(exception=TransactionFailed, exc_text=None):
-        snapshot_id = tester.take_snapshot()
-        with pytest.raises(exception) as excinfo:
+        with env.anchor(), pytest.raises(exception) as excinfo:
             yield excinfo
```

Comment:
do any consumers of tx_failed actually used the yielded excinfo?

Reply:
nope, I can remove it

Reply:
hmm. i think we should not yield it unless it is specifically required, to "hide" the implementation. that also frees up the yield spot for something else in case we need it.